go/cmd/dolt/commands/dump.go

```diff
@@ -0,0 +1,324 @@
+
+	if apr.NArg() > 0 {
+		return nil, errhand.BuildDError("too many arguments").SetPrintUsage().Build()
+	}
+
+	var fileName string
+	resultFormat, _ := apr.GetValue(FormatFlag)
+
+	switch resultFormat {
+	case "", "sql", ".sql":
+		fileName = "doltdump.sql"
+	case "csv", ".csv":
+		//handle CSV filetype
+		//maybe create dir 'doltdump' and put all the csv dump files into it
+	default:
+		return nil, errhand.BuildDError("invalid result format").SetPrintUsage().Build()
+	}
+
+	fileLoc := getExportDestination(fileName)
+
+	return &dumpOptions{
+		format: resultFormat,
+		dest:   fileLoc,
+	}, nil
+}
+
+func getTableArgs(tblName string, destination mvdata.DataLocation) (*tableOptions, errhand.VerboseError) {
+	return &tableOptions{
+		tableName: tblName,
+		src:       mvdata.TableDataLocation{Name: tblName},
+		dest:      destination,
+	}, nil
+}
+
+type DumpCmd struct{}
+
+// Name is returns the name of the Dolt cli command. This is what is used on the command line to invoke the command
+func (cmd DumpCmd) Name() string {
+	return "dump"
+}
+
+// Description returns a description of the command
+func (cmd DumpCmd) Description() string {
+	return "Export all tables in the working set into a file."
+}
+
+// CreateMarkdown creates a markdown file containing the help text for the command at the given path
+func (cmd DumpCmd) CreateMarkdown(wr io.Writer, commandStr string) error {
+	ap := cmd.createArgParser()
+	return CreateMarkdown(wr, cli.GetCommandDocumentation(commandStr, lsDocs, ap))
+}
+
+func (cmd DumpCmd) createArgParser() *argparser.ArgParser {
+	ap := argparser.NewArgParser()
+	ap.SupportsFlag(forceParam, "f", "If data already exists in the destination, the force flag will allow the target to be overwritten.")
+	ap.SupportsString(FormatFlag, "r", "result_file_type", "Define the type of the output file. Valid values are sql and csv. Defaults to sql.")
+
+	return ap
+}
+
+func importStatsCB(stats types.AppliedEditStats) {
+	noEffect := stats.NonExistentDeletes + stats.SameVal
+	total := noEffect + stats.Modifications + stats.Additions
+	displayStr := fmt.Sprintf("Rows Processed: %d, Additions: %d, Modifications: %d, Had No Effect: %d", total, stats.Additions, stats.Modifications, noEffect)
+	displayStrLen = cli.DeleteAndPrint(displayStrLen, displayStr)
+}
+
+// EventType returns the type of the event to log
+func (cmd DumpCmd) EventType() eventsapi.ClientEventType {
+	return eventsapi.ClientEventType_DUMP
+}
+
+// Exec executes the command
+func (cmd DumpCmd) Exec(ctx context.Context, commandStr string, args []string, dEnv *env.DoltEnv) int {
+	ap := cmd.createArgParser()
+	help, usage := cli.HelpAndUsagePrinters(cli.GetCommandDocumentation(commandStr, dumpDocs, ap))
+	apr := cli.ParseArgsOrDie(ap, args, help)
+
+	root, verr := GetWorkingWithVErr(dEnv)
+	if verr != nil {
+		return HandleVErrAndExitCode(verr, usage)
+	}
+
+	tblNames, err := doltdb.GetNonSystemTableNames(ctx, root)
```

Comment:
Move this variable initialization down as close as possible to where it's first used (good general principle)

Reply:
Same applies to all these variables, do a pass and you'll see what I mean